2019/d07Amplifiers.d

```diff
@@ -0,0 +1,131 @@
+auto thrusterSignal(int[] program, int[] phases) {
+  import intcode;
+  int input;
+
+  foreach(phase; phases) {
+    input = IntCode(program.dup).run([phase, input])[0];
```

Comment:
IntCode looks like it needs a more formal specification for input programs and a guarantee of copying the input. Might mean retroactive testing :(